File: C++/midterm/B08504055_2.cpp
//B08504055_2
#include <iostream>
#include <cmath>
using namespace std;

double func(int);

int main(){
	
	int n;
	cout<<"please input a positive number: ";
	cin>>n;
	//(i)
	if(n>=150) cout<<"answer= "<<abs(sin(2*n))<<endl;
	else if(n>=100) cout<<"answer= "<<log10(n)<<endl;
	else cout<<"answer= "<<n<<endl;
	
	//(ii)
	double sum = 1;
	for(int i=1; i<=n; i++){
		sum = sum + pow(-1,i)*pow(0.5,(2*i))/func(2*i);
	}
	cout<<"cos(0.5)= "<<sum<<endl;
	
	return 0;
} 

double func(int x){
	double y = 1;
	for(int j=1; j<=x; j++){
		y = y*j;
	}
	return y;
}

File: MATLAB/midterm/B08504055_2.m
%B08504055_2
clear
clc
% (i)
m = input('please input a positive integer: ');
n = 2;
s = 0;
while s<=m
    x = [1:n];
    s = sum(x.*(x+1).*(x+2));
    n = n + 1;
end
disp(n-1)

% (ii)
A = [1 3 -1;2 -3 3;0 1 1];
B = [3;4;5];
A\B

clear
% (iii)
x = linspace(-1,5,100);
for i=1:100
    xx = x(i);
    if xx<1
        y(i) = abs(xx);
    elseif xx<3
        y(i) = exp(xx)*sin(xx);
    else
        y(i) = log10(xx);
    end
end
plot(x,y)